### DISCUSSION TITLE
PR Checklists

### DISCUSSION THREAD

**Discussion by karimStekelenburg** [openwallet-foundation/credo-ts#914] [Ideas]
Hi all!

With the introduction of the [Aries JavaScript Docs](https://aries.js.org/) and also because I frequently forget to add tests, I'd like to suggest we create a PR template with some default checklist items that need to be ticked off before we merge.

I did a quick search and it looks like we can use [this require checklist action](https://github.com/marketplace/actions/require-checklist) to enforce all items are checked off before a merge is allowed to happen. It simply looks for checklist items in the PR body and blocks if any are unticked. Not sure if this is what we want, because this doesn't allow for optional checkboxes.

This list is definitely incomplete, but as a start:
- Unit tests
- Integration/e2e tests (this should probably be optional, depending on the type of PR)
- Documentation (this should probably be optional, depending on the PR adding/modifying the pubic API)

As always, so suggestions are welcome!

**Comment by 2byrds** [comment]
Good idea!

**Reply by 2byrds** [reply]
Do you want the checklist to include items related to the push and pull request actions that already exist?
For instance `yarn format` is enforced, but should it be a checklist item?

Based on the lessons learned from the 0.2.0 release, what do you think about a checklist item for naming/labeling?  For instance, marking breaking changes with an ! and labeling it with 'breaking change'?  Or maybe that is too specific to include in the checklist.

I don't have experience with massively long checklists for PRs, they could be incredibly helpful or annoying. So, i don't have strong opinions on the amount of effort we should spend to triage what is in the checklist.

**Comment by TimoGlastra** [comment]
Sounds good! I think even the checklist without the enforcement would be helpful, but interested to see how the action will help.

In addition we should probably also create an issue template with the new action forms for bug reports and feature requests. This will push both contributors and users of the framework to submit better issues.

Is this something you'll be willing to open a PR for?

**Comment by TimoGlastra** [comment]
I found this template yesterday which looks pretty good: https://github.com/vitejs/vite/blob/main/.github/ISSUE_TEMPLATE/bug_report.yml. They also have an envinfo command that automatically get's all required system info:

```
npx envinfo --system --npmPackages '{@aries-framework/*,indy-*,react-native,expo}' --binaries --SDKs
```

Will output:

```
  System:
    OS: macOS 12.0.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 220.33 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.13.0/bin/yarn
    npm: 8.1.1 - ~/.nvm/versions/node/v16.13.0/bin/npm
    Watchman: 2022.02.07.00 - /opt/homebrew/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      API Levels: 28, 29, 30, 31
      Build Tools: 28.0.3, 29.0.2, 31.0.0
      System Images: android-29 | Google APIs ARM 64 v8a, android-31 | Google APIs ARM 64 v8a
  npmPackages:
    @aries-framework/core: ^0.2.0-alpha.129 => 0.2.0-alpha.129+8d67e63
    @aries-framework/push-notifications: ^0.2.1 => 0.2.1
    @aries-framework/react-native: ^0.2.0-alpha.129 => 0.2.0-alpha.129+8d67e63
    @aries-framework/redux-store: file:./aries-framework-redux-store-v0.2.0.tgz => 0.2.0
    expo: ^43.0.1 => 43.0.5
    indy-sdk-react-native: ^0.2.0 => 0.2.0
    react-native: 0.64.1 => 0.64.1
```

**Comment by TimoGlastra** [comment]
Getting even further from the initial topic discussion (but also usign it for the issue template part) I created this reproduction template that we can request for bugs / issues: https://github.com/TimoGlastra/afj-reproduction-template

**Comment by TimoGlastra** [comment]
Closing due to the disucssion being old and inactive
